### PR TITLE
applications: nrf_desktop: Align application with upmerge

### PR DESCRIPTION
**Pull request by MarekPieta**
PR aligns nRF Desktop application with the ongoing Zephyr upmerge.

### CODE REVIEW

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf [hunk 1]
```diff
@@ -33,6 +33,7 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 
 # MCUBoot serial
 CONFIG_GPIO=y
+CONFIG_GPIO_NRFX_INTERRUPT=n
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf:36]: Consider migrating to this configuration with all MCUboot projects in the nRF Desktop as we don't use this feature. We should aspire to have uniform behaviour across all targets as we are unable to manually validate the functionality with every target

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/child_image/mcuboot/prj_mcuboot_smp.conf [hunk 2]
```diff
@@ -36,3 +36,6 @@ CONFIG_CBPRINTF_NANO=y
 CONFIG_TIMESLICING=n
 CONFIG_ARM_MPU=n
 CONFIG_THREAD_STACK_INFO=n
+
+# Use minimal C library instead of the Picolib
+CONFIG_MINIMAL_LIBC=y
```
Comment (kapi-no) [inline applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/child_image/mcuboot/prj_mcuboot_smp.conf:41]: Please make sure that you did not skip any of the child image projects

Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/child_image/mcuboot/prj_mcuboot_smp.conf:41]: I think I updated them all (I double checked). Still please double check and notify me if I missed anything.

File: applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf:36]: The GPIO driver is used only for configurations that enable MCUboot serial recovery. Did I overlook any of these configurations?

Comment (kapi-no) [reply applications/nrf_desktop/configuration/nrf52820dongle_nrf52820/child_image/mcuboot/prj_release.conf:36]: Oh, I see. I wasn't aware that only the serial recovery configurations use it. In that case, I have no comments here

**Review by kapi-no — APPROVED** [review]

**Review by mkapala-nordic — APPROVED** [review]

**Comment by MarekPieta** [comment]
The commits were already cherry-picked. I close the ticket.

**Pull request closed without merging**